tests/v2/validation/auth/userretention/userretention_disable_user_test.go

```diff
@@ -0,0 +1,303 @@
+//go:build (validation || infra.any || cluster.any || sanity) && !stress && !extended
+
+package userretention
+
+import (
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/v2/actions/auth"
+	"github.com/rancher/rancher/tests/v2/actions/rbac"
+	"github.com/rancher/shepherd/clients/rancher"
+	managementv3 "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type URDisableTestSuite struct {
+	suite.Suite
+	client      *rancher.Client
+	session     *session.Session
+	adminClient *rancher.Client
+	adminUser   *managementv3.User
+}
+
+func (ur *URDisableTestSuite) SetupSuite() {
+	logrus.Info("Setting up URDisableTestSuite")
+	ur.session = session.NewSession()
+
+	logrus.Info("Creating new Rancher client")
+	client, err := rancher.NewClient("", ur.session)
+	require.NoError(ur.T(), err)
+	ur.client = client
+
+	err = updateUserRetentionSettings(ur.client, authUserSessionTTLMinutes, "0")
+	require.NoError(ur.T(), err)
+}
+
+func (ur *URDisableTestSuite) TearDownSuite() {
+	logrus.Info("Tearing down URDisableTestSuite")
+	if ur.adminUser != nil {
+		logrus.Info("Deleting admin user")
+		err := ur.client.Management.User.Delete(ur.adminUser)
+		if err != nil {
+			logrus.Errorf("Failed to delete admin user: %v", err)
+		}
+	}
+	logrus.Info("Cleaning up session")
+	ur.session.Cleanup()
+}
+
+func (ur *URDisableTestSuite) assertBindingsEqual(before, after map[string]interface{}) {
+	assert.Equal(ur.T(), before["RoleBindings"], after["RoleBindings"], "RoleBindings mismatch")
```

Comment:
Could you replace assert with require in all the tests in the PR? While assert is effective for table-driven tests, these tests are not table-driven